src/ecdsa/ecdh.py

```diff
@@ -0,0 +1,120 @@
+"""
+Class for performing Elliptic-curve Diffie-Hellman (ECDH) operations.
+"""
+
+from .util import number_to_string
+from .ellipticcurve import INFINITY
+from .keys import SigningKey, VerifyingKey
+
+
+__all__ = ["ECDH", "NoKeyError", "InvalidCurveError",
+           "InvalidSharedSecretError"]
+
+
+class NoKeyError(AssertionError):
```

Comment:
why `AssertionError`?

Reply:
```
class NoKeyError(AttributeError):
class NoCurveError(AttributeError):
class InvalidCurveError(RuntimeError):
class InvalidSharedSecretError(RuntimeError):
```

ok?

Reply:
others are inheriting from specific exceptions to preserve API (the code previously was raising AssertionError), but given this is new code, it can get it right the first time over